crates/forge/src/warn.rs

```diff
@@ -0,0 +1,236 @@
+                ignored: false,
+                fork_config: Some(RawForkConfig::Params(RawForkParams {
+                    url: url.into(),
+                    block_id_type: String::new(),
+                    block_id_value: String::new(),
+                })),
+            }],
+        })
+    }
+
+    async fn handler(Json(input): Json<Value>) -> (StatusCode, String) {
+        let id = input.as_object().unwrap().get("id");
+
+        (
+            StatusCode::OK,
+            json!({
+                "id": id,
+                "result": "0.5.0"
+            })
+            .to_string(),
+        )
+    }
+
+    fn setup_fake_node(address: impl Into<String>) {
+        let address = address.into();
+
+        tokio::spawn(async {
+            let app = Router::new().route("/rpc", post(handler));
+
+            let listener = tokio::net::TcpListener::bind(address).await.unwrap();
+            axum::serve(listener, app).await.unwrap();
+        });
+    }
+
+    // must be run with --nocapture or will fail
+    #[tokio::test(flavor = "multi_thread")]
+    async fn should_print_warning() {
+        setup_fake_node("127.0.0.1:3030");
+
+        let test_crates = prepare_input(&["http://127.0.0.1:3030/rpc"]);
+        let buffer = BufferRedirect::stdout().unwrap();
+
+        warn_if_incompatible_rpc_version(&test_crates, &[])
+            .await
+            .unwrap();
+
+        let stdout = read_to_string(buffer.into_inner()).unwrap();
+
+        assert!(stdout == "[WARNING] The RPC node with url = http://127.0.0.1:3030/rpc has unsupported version = (0.5.0), use node supporting RPC version 0.6.0\n");
+    }
+
+    // must be run with --nocapture or will fail
+    #[tokio::test(flavor = "multi_thread")]
+    async fn should_dedup_urls() {
+        setup_fake_node("127.0.0.1:3030");
+
+        let test_crates =
+            prepare_input(&["http://127.0.0.1:3030/rpc", "http://127.0.0.1:3030/rpc"]);
+        let buffer = BufferRedirect::stdout().unwrap();
+
+        warn_if_incompatible_rpc_version(&test_crates, &[])
+            .await
+            .unwrap();
+
+        let stdout = read_to_string(buffer.into_inner()).unwrap();
+
+        assert!(stdout == "[WARNING] The RPC node with url = http://127.0.0.1:3030/rpc has unsupported version = (0.5.0), use node supporting RPC version 0.6.0\n");
+    }
+
+    // must be run with --nocapture or will fail
+    #[tokio::test(flavor = "multi_thread")]
+    async fn should_print_for_each() {
+        setup_fake_node("127.0.0.1:3030");
+        setup_fake_node("127.0.0.1:3035");
+
+        let test_crates =
+            prepare_input(&["http://127.0.0.1:3030/rpc", "http://127.0.0.1:3035/rpc"]);
+        let buffer = BufferRedirect::stdout().unwrap();
+
+        warn_if_incompatible_rpc_version(&test_crates, &[])
+            .await
+            .unwrap();
+
+        let stdout = read_to_string(buffer.into_inner()).unwrap();
+
+        //TODO use assert_stdout_contains!()
```

Comment:
Is this getting fixed?

Reply:
Tag an issue here if u want to do it later on

Reply:
We should fix it right away. Please do not create issues for such cases because no one will fix them

Reply:
Opened https://github.com/foundry-rs/starknet-foundry/issues/1666 for this